Ambit/Source/Ambit/Mode/ConfigImportExport.h

```diff
@@ -36,85 +37,89 @@ struct FVehicleTraffic;
  * Class dedicated to controlling the importing and exporting of configuration files for the AmbitSpawners.
  */
 UCLASS()
-class UConfigImportExport : public UObject
+class UConfigImportExport : public UObject, public IConfigImportExportInterface
 {
     GENERATED_BODY()
 public:
     //Constructor
     UConfigImportExport();
 
-    // SDF import and export
-    /**
-     * Starts the process to import an SDF file specified on screen.
-     *
-     * @return FReply When the process has finished.
-     */
+    /** @inheritDoc */
     FReply OnImportSdf();
 
-    /**
-     * Starts the process to export the SDF file specified. This process triggers asynchronous functions
-     * to handle the exporting, and does not actually export itself.
-     *
-     * @return FReply::Finished When the process has been started for the asynchronous handlings.
-     */
+    /** @inheritDoc */
     FReply OnExportSdf();
 
-    /**
-     * Based on the config settings from screen or the queued config settings from QueuedSdfConfigToExport,
-     * will combine those with the passed AmbitSpawnerArray into a standardized SDF format, and then
-     * writes the file.
-     * This function will attempt to create the S3 bucket, if bToS3 is selected.
-     * At the end of this function, if there are any more items in QueuedSdfConfigToExport, it will call
-     * PrepareAllSpawnersObjectConfigs to continue to the process of the queue.
-     *
-     *
-     * @param AmbitSpawnerArray A Mapping of FJsonValue (coming from SpawnerObjects), and the name of their JSON
-     * format to output for their location in the array.
-     * @param bToS3 Specifies whether the file should attempt to upload to S3 or save to disk.
-     *
-     * @return Boolean to indicate if the write was successful or not.
-     */
+    /** @inheritDoc */
     bool ProcessSdfForExport(const TMap<FString, TSharedPtr<FJsonObject>>& AmbitSpawnerArray, bool bToS3);
 
-    // Generate Permutations
+    /** @inheritDoc */
+    FReply OnImportBsc();
+
+    /** @inheritDoc */
+    FReply OnGeneratePermutations();
+
+    /** @inheritDoc */
+    FReply OnReadFromS3Bucket();
+
+    /** @inheritDoc */
+    FReply OnExportMap();
+
+    /** @inheritDoc */
+    FReply OnExportGltf();
+
     /**
-     * Imports on a BSC file on disk.
+     * Set the glTF exporter to its mocked instance.
      *
-     * @return FReply::Finished When the process has finished.
+     * @param MockExporter The mocked instance of the glTF Exporter.
      */
-    FReply OnImportBsc();
+    void SetDependencies(IGltfExportInterface* MockExporter);
```

Comment:
The change you've mad here doesn't fully address my previous comment. Although you changed the method name so that it no longer refers to mocks, the method's argument and the doc comment still reference mocks. 

Think of this method as a way to plug in any new glTF exporter implementation. So I'd expect it to look like...
```
/**
 * Assign new dependency instances for use by this ConfigImportExport object.
 *
 * @param Exporter The glTF exporter to be used.
 */
void SetDependencies(IGltfExportInterface* Exporter);
```

Reply:
I  see. This makes it clearer. Changed in ConfigImportExport and GltfExport since those are the only two objects that have this function.